### PR TITLE
Added DuffleToolInstaller Task

### PR DESCRIPTION
**Pull request by vithati**

### CODE REVIEW

**Comment by kmkumaran** [comment]
Can you ensure the codeowners file is updated with the new tasks that are being introduced.
https://github.com/Microsoft/azure-pipelines-tasks/blob/master/.github/CODEOWNERS

File: Tasks/DuffleInstallerV0/src/duffleinstaller.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    return path.join(configDir, DuffleToolName + getExecutableExtension());
+}
+
+function getTempDirectory(): string {
+    return tl.getVariable('agent.tempDirectory') || os.tmpdir();
+}
+
+function getExecutableExtension(): string {
```
Comment (madhuv-msft) [inline Tasks/DuffleInstallerV0/src/duffleinstaller.ts]: common library

File: Tasks/DuffleInstallerV0/task.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+    "id": "11EA7CF0-8A3E-4B5A-98A2-8756CC3094B0",
+    "name": "DuffleInstaller",
+    "friendlyName": "Duffle tool installer",
+    "description": "Duffle is a command line tool that allows you to install and manage CNAB bundles",
+    "helpMarkDown": "[More Information](https://go.microsoft.com/fwlink/?linkid=851275)",
```
Comment (madhuv-msft) [inline Tasks/DuffleInstallerV0/task.json]: this link is not updated to point to right URL

File: Tasks/DuffleInstallerV0/src/duffleinstaller.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import path = require('path');
+import fs = require('fs');
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+import * as os from 'os';
+import * as util from 'util';
+import { WebRequest, sendRequest } from 'utility-common/restutilities';
+import { download } from 'utility-common/downloadutility';
+
+const DuffleToolName = 'duffle';
+const DuffleLatestReleaseUrl = 'https://api.github.com/repos/deislabs/Duffle/releases/latest';
```
[the remark this answers is not in the document]
Comment (vithati) [reply Tasks/DuffleInstallerV0/src/duffleinstaller.ts]: Yes. I will move that code to common and will use here once it checked-in.

File: Tasks/DuffleInstallerV0/package.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "dependencies": {
+    "@types/node": "^6.0.101",
+    "@types/q": "^1.5.0",
```
[the remark this answers is not in the document]
Comment (vithati) [reply Tasks/DuffleInstallerV0/package.json]: dependencies.

**Pull request merged**